src/interfaces/commands/command-handler.interface.ts

```diff
@@ -1,5 +1,5 @@
 import { ICommand } from './command.interface';
 
 export interface ICommandHandler<T extends ICommand> {
-    execute(command: T, resolve: (value?) => void);
-}
\ No newline at end of file
+    execute(command: T): Promise<any>;
```

Comment:
What about user generic return type?

`execute<R>(command: T): Promise<R>;`

Reply:
isnt that an overkill? @kamilmysliwiec what you think?

Reply:
i mean that making `handler.execute` generic would imply that we need to carry that generic type definition all the way up to command bus, otherwise well lose that type information in 
` commandbus.execute<T extends ICommand>(command: T): Promise<any> `

Reply:
or.. https://github.com/sh3d2/cqrs/pull/1/files

Reply:
When using generics you don't need to cast the results before using it. There is one small issue. How you know what is return type? You only send command but you don't know the handler. What if the ICommand knows the return type? Using the ICommandHandler should be easy and hide the actually implementation of the handlers.

Reply:
@Matonen i dont really understand the idea of 
>  the ICommand knows the return type

Could you provide some example code?

Reply:
I was just a thought.  

Here is dummy example how i would to like to use execute using the generics:
```
const dto = await commandbus.execute<UserDto>(new CreateUserCommand('John'));
await commandbus.execute<void>(new DeleteUserCommand(dto.id));
```

Here i need to know that CreateUserCommand returns UserDto and DeleteUserCommand does return nothing. How to document that so you don't need to check that from the handler implementation? 

Is it possible to have something like this? 
```
export interface ICommand<ResultT> {}
```